### DISCUSSION TITLE
Is there a possibility to having a similar mechanism to `OriginalUri` but for the `MatchedPath` for nested routers?

### DISCUSSION THREAD

**Discussion by MidasLamb** [tokio-rs/axum#1651] [Q&A] [answered]
The `OriginalUri` feature allows you to extract the originally fully matched `Uri`, regardless of how nested a router is. There is (currently) no such alternative for `MatchedPath`. For otel tracing this is a bit unfortunate because you lose the low cardinality `http.route` attribute, and have to rely on `http.target` for these then.
Is there a possibility to add such a feature flag for `MatchedPath` in nested routers, or is it technically infeasible?

**Comment by davidpdrsn** [comment, marked as answer]
You mean this https://github.com/tokio-rs/axum/issues/1441?
